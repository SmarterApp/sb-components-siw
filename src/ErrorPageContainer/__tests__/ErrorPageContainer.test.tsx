import * as React from "react";
import * as TestUtils from "react-dom/test-utils";
import { shallow } from "enzyme";
import { ErrorPageContainer } from "@src/index";
import {
  serverErrMock,
  notFoundErrMock
} from "@mocks/ErrorPageContainer/mocks";

describe("ErrorPageContainer", () => {
  it("default render empty props", () => {
    const wrapper = shallow(<ErrorPageContainer />);
    expect(wrapper).toMatchSnapshot();
  });

  it("render server error page", () => {
    const wrapper = shallow(<ErrorPageContainer {...serverErrMock} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("render not found page", () => {
    const wrapper = shallow(<ErrorPageContainer {...notFoundErrMock} />);
    expect(wrapper).toMatchSnapshot();
  });
});
