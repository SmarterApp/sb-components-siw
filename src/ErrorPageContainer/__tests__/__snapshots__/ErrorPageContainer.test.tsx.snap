// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPageContainer default render empty props 1`] = `
<div
  className="error-page-container"
>
  <div
    className="error-page-header"
  >
    <h1>
      Sorry!
    </h1>
    <h4>
      <strong>
        Error: 
      </strong>
      Something went wrong
    </h4>
  </div>
  <div
    className="error-page-body"
  >
    <p>
      The page you’re looking for can’t be found. Try searching or returning home!
    </p>
  </div>
</div>
`;

exports[`ErrorPageContainer render not found page 1`] = `
<div
  className="error-page-container"
>
  <div
    className="error-page-header"
  >
    <h1>
      Not Found: 404
    </h1>
    <h4>
      <strong>
        Error: 
      </strong>
      This item cannot be found
    </h4>
  </div>
  <div
    className="error-page-body"
  >
    <p>
      Please try another search
    </p>
  </div>
</div>
`;

exports[`ErrorPageContainer render server error page 1`] = `
<div
  className="error-page-container"
>
  <div
    className="error-page-header"
  >
    <h1>
      Server Error: 500
    </h1>
    <h4>
      <strong>
        Error: 
      </strong>
      you did a bad thing that caused errors
    </h4>
  </div>
  <div
    className="error-page-body"
  >
    <p>
      stop doing that thing that caused errors
    </p>
  </div>
</div>
`;
